src/program.js

```diff
@@ -251,6 +251,12 @@ Example: $0 --help run.
         demand: false,
         type: 'boolean',
       },
+      'pref': {
+        describe: 'Launch firefox with custom preferences. Lightweight ' +
+                  'alternative to creating custom profile.',
+        demand: false,
+        type: 'object',
```

Comment:
But besides the command line syntax, using the `coerce` method seems like a good idea to me (and it is also in a better position than what I suggested in one of my comment above, it is as near as possible to where the external input enters our system), the code in `src/program.js` will become something like:

```
'pref': {
  describe: 'Launch firefox with custom preferences...',
  demand: false,
  requiresArg: true,
  type: 'string',
  coerce: coerceCLICustomPreference,
}
```

And in the helper function we can incapsulate all the details related to validate and parse the command line option, e.g. if we opt for the syntax that I was suggesting above:

- if value received does not represent a valid preference, we can `throw new UsageError("Invalid Preference...");` and web-ext will not run the command and it will show the usage help and the error message raised from our helper
  
- we can split key and value and return an object, e.g.
``` 
  let [key, value] = value.split("=");

  // ... detect and convert integer and boolean values, 
  // and fallback to a string value 

  // The following is a weird but useful ES6 syntax for object
  // property names which are computed:
  return {[`${key}`]: value};
```

For the coercion of the value, what I'm thinking of is not too far from the above (there are some typos in the snippet that you pasted above ;-) but no worries, we will get into the details once we are reviewing the actual commit, and typos are pretty common when sketching example code in a comment), e.g. to detect and convert an integer, I was thinking of something like:

```
if ( value === `${parseInt(value)}`) {
  value = parseInt(value, 10);
}
```

Because we want to be sure that we only convert the value to a number if it is an integer (and `parseInt("123.123", 10)` returns `123` and ignore the rest, and `parseInt` will try to detect the radix if it is not specified explicitly, you can find more details about the `parseInt` behavior on https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt).